docs/compute_resources.md

```diff
@@ -0,0 +1,173 @@
+which nodes to place pods on, and contention for resources can be handled in a consistent
+manner.
+
+
+*CPU* and *memory* are each a *resource type*.  A resource type has a base unit.
+CPU is specified in units of cores.  Memory is specified in units of bytes.
+
+CPU and RAM are collectively refered to as *compute resources*, or just *resources*.  Compute
+resources are measureable quantities which can be requested, allocated, and consumed.  They are
+distinct from [API resources](./working_with_resources.md).  API resources, such as pods and
+[services](./services.md) are objects that can be written to and retrieved from the Kubernetes API
+server.
+
+## Container and Pod Resource Limits
+Each container of a Pod can optionally specify `spec.container[].resources.limits.cpu` and/or
+`spec.contawiner[].resources.limits.memory`.  The `spec.container[].resources.requests` field
+is not currently used and need not be set.
+
+Specifying resource limits is optional.  In some clusters, an unset value may be replaced with a
+default value when a pod is created or updated.  The default value depends on how the cluster is
+configured.
+
+Although limits can only be specified on individual containers, it is convenient to talk about pod
+resource limits.
+A *pod resource limit* for a particular resource type is the sum of the resource limits of that type for each
+container in the pod, with unset values treated as zero.
+
+The following pod has two containers.  Each has a limit of 0.5 core of cpu and 100MB of memory.
+The pod can be said to have a limit of 1 core and 200MB of memory.
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: frontend
+spec:
+  containers:
+  - name: db
+    image: mysql
+    resources:
+      limits:
+        memory: "100M"
+        cpu: "500m"
+  - name: wp
+    image: wordpress
+    resources:
+      limits:
+        memory: "100M"
+        cpu: "500m"
+```
+
+## How Pods with Resource Limits are Scheduled
+
+When a pod is created, the kubernetes scheduler selects a node for the pod to run on.  A node has a
+capacity: the amount of CPU and memory it can provide for pods.  The scheduler ensures that, for
+each resource type (CPU and memory), the sum of the resource limits of the containers scheduled to
+the node is less than the capacity of the node.  Note that although actual memory or CPU resource
+usage on nodes is very low, the scheduler will still refuse to place pods onto nodes if the capacity
+check fails.  This protects against a resource shortage on a node when resource usage later increases,
+such as due to a daily peak in request rate.
+
+If a pod does not specify a resource limit, then the scheduler treats the limit as zero.
+The scheduler also tries to spread pods evenly across nodes, among other scheduling
+considerations.  As a result, even if pods do not specify resource limits, the system may
+still function fine.  
```

Comment:
> I guess since the addons have limits now, you won't have the case where no pods have limits.

Yeah.

> Do we really want usage aware scheduler? Why not a vertical autoscaler adjusting limits, and then the scheduler continues to schedule based on limits? Seems like a better decoupling of system components.

Yes, my wording was meant to be a summary, not a design proposal :) I agree it should be implemented via vertical autopilot type thingy.